sdks/python/apache_beam/coders/typecoders.py

```diff
@@ -119,15 +119,21 @@ def register_coder(self, typehint_type, typehint_coder_class):
     if typehint_type.__module__ == '__main__':
       # See https://issues.apache.org/jira/browse/BEAM-14250
       # TODO(robertwb): Remove once all runners are portable.
-      typehint_type = str(typehint_type)
+      if hasattr(typehint_type, '__name__'):
+        typehint_type = typehint_type.__name__
+      else:
+        typehint_type = str(typehint_type)
     self._register_coder_internal(typehint_type, typehint_coder_class)
 
   def get_coder(self, typehint):
     # type: (Any) -> coders.Coder
     if typehint and typehint.__module__ == '__main__':
       # See https://issues.apache.org/jira/browse/BEAM-14250
       # TODO(robertwb): Remove once all runners are portable.
-      typehint = str(typehint)
+      if hasattr(typehint, '__name__'):
```

Comment:
`getattr(typehint, '__name__', str(typehint))`

Similarly above.

Reply:
Thanks, will apply it!